python/ray/train/tests/test_session.py

```diff
@@ -165,17 +166,20 @@ def encode_checkpoint(checkpoint):
     def validate_encoded():
         next = session.get_next()
         assert next is not None
-        assert next.type == TrainingResultType.CHECKPOINT
         assert next.data["encoded"] is True
 
     init_session(
         training_func=train_func,
         world_rank=0,
         local_rank=0,
         world_size=1,
-        encode_checkpoint_fn=encode_checkpoint)
+        encode_data_fn=encode_checkpoint)
+
     session = get_session()
     session.start()
+    # Validate checkpoint is encoded.
+    validate_encoded()
+    # Validate report is encoded.
     validate_encoded()
```

Comment:
nit: To make this more explicit
```suggestion
    # Validate checkpoint is encoded.
    validate_encoded(TrainingResultType.CHECKPOINT)
    # Validate report is encoded.
    validate_encoded(TrainingResultType.REPORT)
```

and update `validate_encoded`:

```python
    def validate_encoded(type: TrainingResultType):
        next = session.get_next()
        assert next is not None
        assert next.type == type
        assert next.data["encoded"] is True
```

Reply:
Done!